src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/rulemanager/OfBatchHolder.java

```diff
@@ -146,11 +143,7 @@ public BatchData getByUUid(UUID uuid) {
      * Get speaker data by UUID.
      */
     public SpeakerData getSpeakerDataByUUid(UUID uuid) {
-        return Stream.of(flowsMap.values(), metersMap.values(), groupsMap.values())
-                .flatMap(Collection::stream)
-                .filter(data -> uuid.equals(data.getUuid()))
-                .findFirst()
-                .orElseThrow(() -> new IllegalStateException(format("Can't find speaker data with uuid %s", uuid)));
+        return commandMap.get(uuid).getOrigin();
```

Comment:
Is it possible to get NPE here?

Reply:
No, commandMap should contains all uuids present in the batch and this method is called for all uuids in the batch.